lib/doorkeeper/models/concerns/revocable.rb

```diff
@@ -8,6 +8,27 @@ def revoke(clock = Time)
       def revoked?
         !!(revoked_at && revoked_at <= Time.now)
       end
+
+      def revoke_in(time)
+        update_attribute :revoked_at, Time.now + time
+      end
+
+      def revoke_previous_refresh_token!
+        if old_refresh_token && !old_refresh_token.revoked?
+          old_refresh_token.revoke_in(refresh_token_revoked_in)
+        end
+        if previous_refresh_token.present?
+          update_attribute :previous_refresh_token, ""
+        end
```

Comment:
I think we don't need this conditional, and we can always execute line 21. Would replace these three lines by `update_attribute :previous_refresh_token, ""`.